engines/sci/graphics/text32.cpp

```diff
@@ -364,7 +377,22 @@ void GfxText32::drawText(const uint index, uint length) {
 	// This draw loop implementation is somewhat different than the
 	// implementation in SSCI, but is accurate. Primarily the changes revolve
 	// around eliminating some extra temporaries and fixing the logic to match.
-	const char *text = _text.c_str() + index;
+
+	Common::String textString;
+	const char *text;
+	if (g_sci->getLanguage() == Common::HE_ISR) {
+#ifdef USE_TRANSLATION
```

Comment:
That is incorrect and will not work when translation is enabled but BiDi is not. You have to wrap it in both conditions.

Reply:
But see in `common\translation.cpp`:

```
#ifdef USE_FRIBIDI
String TranslationManager::convertBiDiString(const String &input, const Common::Language lang) {

...

}
#else
String TranslationManager::convertBiDiString(const String &input, const Common::Language lang) {
	return input;
}
#endif
```

Therefore, in case that translation is enabled but BiDi is not, it should just return the original input.

If you're worried about the `(g_sci->getLanguage() == Common::HE_ISR)` - it's OK, because `SciEngine::getLanguage()` is always defined, and not dependent on any `ifdef`.

Anyway, I have just verified it without `USE_TRANSLATION`, and it worked well.

Reply:
Maybe the `convertBiDiString()` that takes a language could be moved out of `TranslationManager` so that it is available also when `USE_TRANSLATION` is not defined?

I am also wondering if assuming `iso8859-8` is a good idea, and maybe we might want to pass the charset to the function as well (e.g. maybe we support some Hebrew games that use Unicode encoding and would benefit from this, or we might want to change TranslationManager to use Unicode in the future but without breaking BiDi in SCI).

Reply:
@criezy , I agree.
Let's wait with this PR until #2217 is approved, and then I'll update the code here.

Reply:
For a second there I misread that as "SCI games that use Unicode encoding" and thought someone had beaten me to it.